From 2765924829dc34915ed32174647d638a00db1cb0 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Wed, 8 Jan 2020 09:53:59 +0800
Subject: [PATCH] drm/bridge: synopsys: dw-hdmi: reset HDR metadata before
 getting edid

HDR panel metadata is in connector, not in drm_hdmi_info. So HDR
panel metadata won't be reset when parsing edid. If hdmi is connected
from one HDR TV to another TV that does not support HDR, HDR panel
metadata is still from HDR TV, that will cause hdmi display error.

Change-Id: I2c38240cdb8a7ab6517fa5987db21bc28940da38
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
index 06d6a9953e2f..53a3f1aa110a 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -2610,6 +2610,7 @@ static int dw_hdmi_connector_get_modes(struct drm_connector *connector)
 	struct drm_display_info *info = &connector->display_info;
 	int i, ret = 0;
 
+	memset(metedata, 0, sizeof(*metedata));
 	if (!hdmi->ddc)
 		return 0;
 
-- 
2.35.3

